src/Validation/Validator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Validation;
+
+use Lcobucci\JWT\Token;
+
+/**
+ * @author Luís Otávio Cobucci Oblonczyk <lcobucci@gmail.com>
+ *
+ * @since 4.0.0
+ */
+final class Validator implements \Lcobucci\JWT\Validator
```

Comment:
`InvalidTokenException` is not a `ConstraintViolationException` and my idea is that users should only know about `InvalidTokenException`.

Of course we could turn these into a composite specification (which is closer to what you're describing IMO) but I don't think we need that complexity...

> A validator simply returns true/false

I'm a bit skeptical about it... not saying that's not true but it sounds over simplified.
Maybe we can offer both things:

```php
interface Validator
{
    /**
     * @throws InvalidTokenException
     */
    public function assert(Token $token, Constraint ...$constraints): void;
    public function validate(Token $token, Constraint ...$constraints): bool;
}
```

One would use `assert()` when is looking for a "detailed report" or `validate()` when is just caring about the end result.